docs/core/diagnostics/diagnostic-libraries.md

```diff
@@ -0,0 +1,161 @@
+
+The <xref:Microsoft.Extensions.Diagnostics.ExceptionSummarization.IExceptionSummarizer.Summarize%2A?displayProperty=nameWithType> method systematically traverses the roster of registered summarizers until it identifies a summarizer capable of handling the specific exception type. In the event that no summarizer is capable of recognizing the exception type, a meaningful default exception summary is provided instead.
+
+### Example exception summarization usage
+
+The following example demonstrates how to use the `IExceptionSummarizer` interface to retrieve a summary of an exception.
+
+:::code source="snippets/exception-summary/Program.cs":::
+
+The preceding code:
+
+- Instantiates a new <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> instance, chaining a call to the <xref:Microsoft.Extensions.Diagnostics.ExceptionSummarization.ExceptionSummarizationExtensions.AddExceptionSummarizer%2A> extension method.
+  - The `AddExceptionSummarizer` extension method accepts a delegate that is used to configure the `ExceptionSummarizerBuilder` instance.
+  - The `builder` is used to add the HTTP provider, which handles exceptions of type:
+    - <xref:System.OperationCanceledException>
+    - <xref:System.Threading.Tasks.TaskCanceledException>
+    - <xref:System.Net.Sockets.SocketException>
+    - <xref:System.Net.WebException>
+- Builds a new `ServiceProvider` instance from the `ServiceCollection` instance.
+- Gets an instance of the `IExceptionSummarizer` interface from the `ServiceProvider` instance.
+- Iterates over a collection of exceptions, calling the `Summarize` method on each exception and displaying the result.
+
+> [!WARNING]
+> The primary focus in the design of all exception summarization implementations is to provide diagnostic convenience to trusted users, rather than prioritizing the protection of personally identifiable information (PII). The <xref:Microsoft.Extensions.Diagnostics.ExceptionSummarization.ExceptionSummary.Description?displayProperty=nameWithType> doesn't contain sensitive information, but the <xref:Microsoft.Extensions.Diagnostics.ExceptionSummarization.ExceptionSummary.AdditionalDetails?displayProperty=nameWithType> might contain sensitive information depending on the implementation.
+
+## Resource monitoring
+
+Resource monitoring involves the continuous measurement of resource utilization over a specified period. The [Microsoft.Extensions.Diagnostics.ResourceMonitoring](https://www.nuget.org/packages/Microsoft.Extensions.Diagnostics.ResourceMonitoring) NuGet package offers a collection of APIs tailored for monitoring the resource utilization of your .NET applications.
+
+The <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.IResourceMonitor> interface furnishes methods for retrieving real-time information concerning process resource utilization. This interface supports the retrieval of data related to CPU and memory usage and is currently compatible with both Windows and Linux platforms.
```

Comment:
I think it is worth to mention, that this information is published to OTel by default, so one does not need to publish it on themselves.